packages/hdwallet-native/src/bitcoin.ts

```diff
@@ -269,7 +271,10 @@ export function MixinNativeBTCWallet<TBase extends core.Constructor<NativeHDWall
                 address = bchAddr.toLegacyAddress(address);
               }
 
-              psbt.addOutput({ address, value: Number(amount) });
+              // OP_RETURN_DATA output is not part of the outputs just yet, it gets added below in the `if (msg.opReturnData)` block
+              // So missing amount *is* a blatant disaster waiting to happen here
```

Comment:
Pretty much what the comment says, `amount` can now be undefined which is invalid as a `BigInt` arg
![Uploading image.png…]()